lib/streamlit/watcher/event_based_path_watcher.py

```diff
@@ -181,9 +205,20 @@ def close(self) -> None:
 class WatchedPath(object):
     """Emits notifications when a single path is modified."""
 
-    def __init__(self, md5, modification_time):
+    def __init__(
+        self,
+        md5,
+        modification_time,
+        *,  # keyword-only arguments:
+        glob_pattern: Optional[str] = None,
+        allow_nonexistent: bool = False,
+    ):
```

Comment:
I know it started out without them , but could we add type annotations where they're missing here?

Reply:
👍 can do